src/components/Microsoft.DotNet.UpgradeAssistant/Checks/MultiTargetFrameworkCheck.cs

```diff
@@ -29,24 +29,33 @@ public MultiTargetFrameworkCheck(ILogger<MultiTargetFrameworkCheck> logger)
 
         public string UpgradeMessage => string.Format(CultureInfo.InvariantCulture, "Please see {0} to request this feature.", FEATURE_LINK);
 
-        public Task<UpgradeReadiness> IsReadyAsync(IProject project, UpgradeReadinessOptions options, CancellationToken token)
+        public async Task<UpgradeReadiness> IsReadyAsync(IProject project, UpgradeReadinessOptions options, CancellationToken token)
         {
             if (project is null)
             {
                 throw new ArgumentNullException(nameof(project));
             }
 
             var tfms = project.TargetFrameworks;
+            var components = await project.GetComponentsAsync(token).ConfigureAwait(false);
 
             if (tfms.Count == 1)
             {
                 _logger.LogTrace("Confirmed project {Project} has a valid TFM ({TFM})", project.FileInfo, tfms.First());
-                return Task.FromResult(UpgradeReadiness.Ready);
+                return UpgradeReadiness.Ready;
             }
             else
             {
-                _logger.LogError("Project {Project} cannot be upgraded. Input projects must have exactly one target framework. Please see {FeatureLink} to request this feature.", project.FileInfo, FEATURE_LINK);
-                return Task.FromResult(UpgradeReadiness.NotReady);
+                if (components.HasFlag(ProjectComponents.Maui))
+                {
+                    _logger.LogTrace("Confirmed project {Project} is of type .NET MAUI Head with TFMs : ({TFM}))", project.FileInfo, tfms.First());
```

Comment:
Is the purpose here just to log that there are Maui head projects that are multi-targeting? Coz I see that the return type is still NotReady. Although the messaging is very confusing as it does not clearly tell the customer that they can't move forward coz they are multi-targeting. I would do this instead to make it more obvious: 
` 

else{

               if (components.HasFlag(ProjectComponents.Maui))
                {
                    _logger.LogTrace("Project {Project} is of type .NET MAUI Head with multiple TFMs : ({TFM}))", project.FileInfo, tfms.First());
               }
             _logger.LogError("Project {Project} cannot be upgraded. Input projects must have exactly one target framework. Please see {FeatureLink} to request this feature.", project.FileInfo, FEATURE_LINK);
                return UpgradeReadiness.NotReady;
}

`

Reply:
This part happens when legacy xamarin head project is converted to Maui head project, all steps are complete and then in the finalize step, you see this Log Error that multi target is not supported, I basically wanted to put wording specifically for MAUI projects that "finalized and completed for MAUI head, TFMs are added" and not show the second message because then MAUI migration folks would be confused. Also, set it to NotReady but not Error because migration is done, no more steps are needed if project head is already MAUI style with TFMs. 

So Basically, 
if MAUI == show Information message that migration complete, and move onto next project
else == show error message that Multiple TFMs not supported.

Reply:
IsReadyCheck is call that is made at the beginning of the upgrade in the CurrentProjectSelectionStep, I don't see the readyCheck being called in the FinalizeSolutionStep. Maybe I'm missing something?

Reply:
marking this resolved, after talking to you both, created this issue to further discuss the best path forward : https://github.com/dotnet/upgrade-assistant/issues/797